network/src/external/protocol/ping/ping_pong_worker.rs

```diff
@@ -34,43 +34,38 @@ pub enum PingState {
 /// 2. The peer responds with a Pong message.
 /// 3. The server verifies the Pong message and updates the peer's last seen date
 #[derive(Clone, Debug)]
-pub struct PingProtocol {
+pub struct PingPongWorker {
     state: PingState,
-    channel: Arc<Channel>,
     nonce: u64,
 }
 
-impl PingProtocol {
+impl PingPongWorker {
     /// Send the initial ping message to a peer.
-    pub async fn send(channel: Arc<Channel>) -> Result<Self, PingProtocolError> {
+    pub async fn send(channel: &Arc<Channel>) -> Result<Self, PingProtocolError> {
```

Comment:
`Arc<T>` implements `Deref<Target = T>`, so if it doesn't need to be cloned, just `&T` (in this case `&Channel`) can be put as an argument